Libraries/LibIPC/TransportSocket.h

```diff
@@ -48,8 +50,14 @@ class TransportSocket {
     ErrorOr<void> transfer(ReadonlyBytes, Vector<int, 1> const& unowned_fds);
 
     NonnullOwnPtr<Core::LocalSocket> m_socket;
+    NonnullOwnPtr<Threading::Mutex> m_socket_write_mutex;
```

Comment:
does the mutex need allocated on the heap? storing it in a nnop seems unnecessary at first glance

Reply:
right, using NonnullOwnPtr for mutex is a workaround for TransportSocket being movable while Mutex being non-movable. although now when you mentioned this, I am thinking that probably we better make TransportSocket non-movable too, instead of stacking these hacks.